lang_java/tree_sitter/codegen.py

```diff
@@ -0,0 +1,109 @@
+import json
+
+with open("node-types.json", 'r') as fs:
+    data = json.load(fs)
+
+codegen = '''
+(* 
+ * Yoann Padioleau, Sharon Lin
+ * 2020 initial draft
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(*
+ * An AST for Java.
+ *
+ * todo:
+ *  - finish ast import from 
+ *    https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js
+ *)
+ (*****************************************************************************)
+(* The Tree-sitter AST java related types *)
+(*****************************************************************************)
+(* ------------------------------------------------------------------------- *)
+(* Token/info *)
+(* ------------------------------------------------------------------------- *)
+
+type tok = Parse_info.t
+  (* with tarzan *)
+type 'a wrap  = 'a * tok
+  (* with tarzan *)
+
+type 'a list1 = 'a list (* really should be 'a * 'a list *)
+  (* with tarzan *)
+'''
+
+for a in data:
```

Comment:
ideally, instead of being huge if-elif blobs, each case would indicate what it is doing and will have helper sub functions that handles each case! That way this code is more modular and easy to test